bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsDeviceHandler.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.elroconnects.internal.handler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.elroconnects.internal.devices.ElroConnectsDevice;
+import org.openhab.binding.elroconnects.internal.util.ElroConnectsUtil;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+
+/**
+ * The {@link ElroConnectsDeviceHandler} represents the thing handler for an Elro Connects device.
+ *
+ * @author Mark Herwege - Initial contribution
+ */
+@NonNullByDefault
+public class ElroConnectsDeviceHandler extends BaseThingHandler {
+
+    protected @Nullable Integer deviceId;
+
+    public ElroConnectsDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        ElroConnectsDeviceConfiguration config = getConfigAs(ElroConnectsDeviceConfiguration.class);
+        Integer id = config.deviceId;
+        deviceId = id;
+
+        if (id == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Device ID not set");
+            return;
+        }
```

Comment:
Just as a comment no need to change anything, you could have marked the config.deviceId as a required config in the xml. Then the core will not create your handler unless all the configs that are required have a value. This means not having them marked as Nullable and you don't need to check for it at least 6 times that I have noticed so far.

Reply:
I did some refactoring and avoided many of these checks.